### ISSUE TITLE
`useExhaustiveDependencies`: should warn when `ref.current` is listed in a dependency array

### ISSUE THREAD

**Issue by unional** [biomejs/biome#10685]
## What I'd love to see

ESLint's `react-hooks/exhaustive-deps` rule emits a specific diagnostic when `ref.current` appears in a dependency array:

> Mutable values like 'ref.current' aren't valid dependencies because mutating them doesn't re-render the component.

Biome's `useExhaustiveDependencies` currently recognizes `useRef` results as stable (so it won't ask you to *add* them), but it doesn't warn in the opposite direction — when a developer *includes* `ref.current` in a dependency array. This is a common anti-pattern that silently produces incorrect behavior.

## Why this matters

Including `ref.current` in a dependency array gives the false impression that the effect/memo/callback will re-run when the ref's value changes. It won't, because:

1. Mutating `ref.current` does not trigger a re-render
2. Without a re-render, React never re-evaluates dependency arrays
3. The value captured in the dep array becomes stale

This leads to subtle bugs that are difficult to track down.

## Example

```jsx
function MyComponent({ gridRef }) {
  const [isOpen, setIsOpen] = useState(false)

  const togglePopup = useCallback(() => {
    gridRef?.current?.api?.openToolPanel?.('columns')
    setIsOpen(true)
  }, [gridRef?.current, isOpen]) // ← ref.current in deps: silent bug

  useEffect(() => {
    if (gridRef?.current && isOpen) {
      gridRef.current.api?.setGridOption('sideBar', { /* ... */ })
    }
  }, [gridRef?.current, isOpen]) // ← same issue
}
```

ESLint would flag both dependency arrays. Biome passes silently.

## Expected behavior

Biome should emit a diagnostic like:

> Mutable values like 'gridRef.current' aren't valid dependencies because mutating them doesn't re-render the component. Either exclude it or remove the dependency array.

This should apply to any `.current` property access on a value returned from `useRef()` or typed as `RefObject<T>` / `MutableRefObject<T>` when it appears in a hook dependency array.

## Prior art

- ESLint `react-hooks/exhaustive-deps` — emits this diagnostic since [facebook/react#14930](https://github.com/facebook/react/commit/1bbfbc98d)
- React docs: ["Changing a ref does not trigger a re-render"](https://react.dev/reference/react/useRef)
- @kentcdodds: ["Don't ever do `useEffect(() => {}, [ref.current])`"](https://www.epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array)

## Additional context

The cleanup-phase variant (accessing `ref.current` inside a cleanup function) is a related but separate concern also handled by ESLint's rule. That could be addressed separately or in the same PR.
